.circleci/config.yml

```diff
@@ -44,6 +44,51 @@ jobs:
           working_directory: ./web
           command: yarn test:unit
 
+  test_e2e:
+    docker:
+      - image: circleci/python:3.6-node-browsers
+      - image: viime/opencpu-base
+        command: |
+          /bin/sh -c "chmod a+w /usr/local/lib/R/site-library && service cron start && /usr/lib/rstudio-server/bin/rserver && apachectl -DFOREGROUND"
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - yarn-packages-{{ checksum "e2e-test/package.json" }}
+      - save_cache:
+          name: Save Yarn Package Cache
+          key: yarn-packages-{{ checksum "e2e-test/package.json" }}
+          paths:
+            - ~/.cache/yarn
```

Comment:
This is cool, but premature. Caching is purely for speeding up a slow build by skipping some steps, but this isn't working at all yet. I'd just completely ignore caching until there's a need for it. Personally I'd consider any end to end testing runtime under 5 minutes as perfectly acceptable.
It looks like this configuration wouldn't actually do any useful caching anyway. You restore the cache as the first step (which is fine), then immediately save the cache again without actually installing anything. If anything, I believe the order would be:
* `restore_cache`
* `run: yarn install` (if the cache was loaded, this is fast, otherwise it takes ~2 minutes)
* `save_cache`

Thanks for bringing this up! I'd never heard of this.